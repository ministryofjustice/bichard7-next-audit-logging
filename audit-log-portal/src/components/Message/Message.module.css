.container {
  display: block;
  margin-bottom: 1rem;
}

.inner-container {
  display: flex;
  flex-direction: row;
}

.information-block {
  margin-right: 2rem;
}

.label {
  font-weight: bold;
}
