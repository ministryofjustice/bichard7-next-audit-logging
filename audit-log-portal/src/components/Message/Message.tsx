import DateTime from "components/DateTime"
import classNames from "classnames"
import styles from "./Message.module.css"

interface Props {
  message: {
    messageId: string
    caseId: string
    receivedDate: Date
  }
}

const Message = ({ message }: Props): JSX.Element => (
  <div className={classNames(styles.container, "row")}>
    <h1>123</h1>
    <h1>321</h1>
    <div className={styles["inner-container"]}>
      <div className={styles["information-block"]}>
        <p className={styles.label}>{`Message Id`}</p>
        <p>{message.messageId}</p>
      </div>

      <div className={styles["information-block"]}>
        <p className={styles.label}>{`Case Number`}</p>
        <p>{message.caseId}</p>
      </div>

      <div className={styles["information-block"]}>
        <p className={styles.label}>{`Received Date`}</p>
        <p>
          <DateTime date={message.receivedDate} />
        </p>
      </div>
    </div>
  </div>
)

export default Message
