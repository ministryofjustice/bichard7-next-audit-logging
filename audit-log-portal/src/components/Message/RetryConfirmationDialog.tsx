import styled from "styled-components"
import React from "react"
import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@material-ui/core"
import type { AuditLog } from "shared"
import InformationIcon from "icons/InformationIcon"

export interface SimpleDialogProps {
  open: boolean
  selectedValue: AuditLog
  onClose: (r: boolean) => void
}

const InfoText = styled.div`
  padding-left: 10px;
`
const InfoParagraph = styled.div`
  display: flex;
  align-items: center;
`

export function RetryConfirmationDialog(props: SimpleDialogProps) {
  const { onClose, selectedValue, open } = props

  const triggerRetry = () => {
    /* do nothing for now */
    // add config logic here
    // fetch(`${config.apiUrl}/messages/${selectedValue.messageId}/retry`, {method: 'POST'});
    onClose(true)
  }

  return (
    <Dialog aria-labelledby="retry-dialog-title" open={open}>
      <DialogTitle id="retry-dialog-title">Retry Message</DialogTitle>
      <DialogContent>
        <InfoParagraph>
          <InformationIcon />
          <InfoText>
            {`Are you sure you wish to retry "${selectedValue.externalCorrelationId}" from the last failure?`}
          </InfoText>
        </InfoParagraph>
      </DialogContent>
      <DialogActions>
        <Button onClick={triggerRetry} color="primary" variant="contained">
          {`Yes`}
        </Button>
        <Button onClick={() => onClose(false)}>{`No`}</Button>
      </DialogActions>
    </Dialog>
  )
}
