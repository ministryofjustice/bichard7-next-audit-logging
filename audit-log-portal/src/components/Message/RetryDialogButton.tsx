import { useState } from "react"
import RetryIcon from "icons/RetryIcon"
import { CircularProgress, IconButton, Tooltip } from "@material-ui/core"
import type { AuditLog } from "shared"
import If from "components/If"
import { RetryDialogConfirm } from "./RetryDialogConfirm"

interface Props {
  message: AuditLog
  show: boolean
  isRetrying: boolean
}

export default function RetryDialogButton({ message, show, isRetrying }: Props) {
  const [open, setOpen] = useState(false)

  return (
    <If condition={show}>
      {isRetrying ? (
        <>
          <Tooltip title="Retrying" aria-label="retry">
            <CircularProgress color="secondary" />
          </Tooltip>
        </>
      ) : (
        <>
          <IconButton onClick={() => setOpen(true)}>
            <RetryIcon />
          </IconButton>
          <RetryDialogConfirm selectedValue={message} open={open} onClose={() => setOpen(false)} />
        </>
      )}
    </If>
  )
}
