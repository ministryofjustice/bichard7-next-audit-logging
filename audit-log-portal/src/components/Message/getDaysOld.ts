import { formatDistance } from "date-fns"

export default (date: Date | string): string => {
  const today = new Date()
  const actualDate = new Date(date)

  return formatDistance(actualDate, today, { addSuffix: true }).replace("ago", "old")
}
