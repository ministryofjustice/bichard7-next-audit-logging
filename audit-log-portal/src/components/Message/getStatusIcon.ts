import { ComponentType } from "react"
import ErrorIcon from "icons/ErrorIcon"
import SuccessIcon from "icons/SuccessIcon"
import ProcessingIcon from "icons/ProcessingIcon"
import { AuditLogStatus } from "shared"

export default function getStatusIcon(status: string): ComponentType {
  switch (status) {
    case AuditLogStatus.processing:
      return ProcessingIcon

    case AuditLogStatus.completed:
      return SuccessIcon

    default:
      return ErrorIcon
  }
}
