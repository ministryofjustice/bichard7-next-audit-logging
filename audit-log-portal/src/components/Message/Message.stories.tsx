import { Meta } from "@storybook/react"
import Message from "./Message"

export default {
  title: "components/Message",
  component: Message
} as Meta

export const Default = () => (
  <Message
    message={{
      messageId: "LIBRA-EXISS-001",
      caseId: "Case Id",
      receivedDate: new Date()
    }}
  />
)
