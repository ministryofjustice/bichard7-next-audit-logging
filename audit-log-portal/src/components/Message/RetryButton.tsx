import { useState } from "react"
import RetryIcon from "icons/RetryIcon"
import ProcessingIcon from "icons/ProcessingIcon"
import { IconButton, Tooltip } from "@material-ui/core"
import type { AuditLog } from "shared"
import If from "components/If"
import { RetryConfirmationDialog } from "./RetryConfirmationDialog"

interface Props {
  message: AuditLog
  show: boolean
  isRetrying: boolean
}

export default function RetryButton({ message, show, isRetrying }: Props) {
  const [open, setOpen] = useState(false)

  return (
    <If condition={show}>
      {isRetrying ? (
        <>
          <Tooltip title="Retrying" aria-label="retry">
            <ProcessingIcon />
          </Tooltip>
        </>
      ) : (
        <>
          <Tooltip title="Retry message" aria-label="retry">
            <IconButton onClick={() => setOpen(true)}>
              <RetryIcon />
            </IconButton>
          </Tooltip>
          <RetryConfirmationDialog selectedValue={message} open={open} onClose={() => setOpen(false)} />
        </>
      )}
    </If>
  )
}
