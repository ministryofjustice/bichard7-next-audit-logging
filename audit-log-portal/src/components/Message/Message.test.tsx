import { render, screen } from "@testing-library/react"
import Message from "./Message"

test("renders all 3 values when given a message", () => {
  const message = {
    messageId: "Message1",
    caseId: "Case1000",
    receivedDate: new Date("2021-05-01T10:25:53")
  }

  render(<Message message={message} />)

  expect(screen.getByLabelText("Message Id")).toHaveTextContent(message.messageId)
  expect(screen.getByLabelText("Case Id")).toHaveTextContent(message.caseId)
  expect(screen.getByLabelText("Received Date")).toHaveTextContent("01/05/2021 10:25:53")
})
