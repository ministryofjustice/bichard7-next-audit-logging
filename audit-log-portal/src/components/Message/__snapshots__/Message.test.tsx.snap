// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root Message__Container-sc-12ihk2z-0 jpkXTb MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root Message__InnerContainer-sc-12ihk2z-1 jgWPQx"
    >
      <div
        class="Message__Block-sc-12ihk2z-2 hEUOFb"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          ExternalCorrelationId
        </h6>
        <span
          class="MuiTypography-root Message__ReceivedDate-sc-12ihk2z-3 biKlAM MuiTypography-caption"
        >
          Received: 
          <time
            aria-label="time"
          >
            01/05/2021 10:25:53
          </time>
        </span>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        23 days old
      </h6>
      <div
        class="Message__Actions-sc-12ihk2z-4 cFTIpd"
      >
        <span
          class="MuiBadge-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-calendar-day fa-w-14 "
                  data-icon="calendar-day"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style="width: 18px; height: 18px;"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              View Events
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
          >
            0
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
